src/packaging/tags.py

```diff
@@ -92,6 +93,13 @@ def __str__(self) -> str:
     def __repr__(self) -> str:
         return f"<{self} @ {id(self)}>"
 
+    def __setstate__(self, state: tuple[dict[str, Any], dict[str, Any]]) -> None:
+        # cached _hash is wrong when unpickling
+        _, slots = state
```

Comment:
```suggestion
        # The cached _hash is wrong when unpickling.
        _old_hash, slots = state
```

Reply:
I'm fixing it in https://github.com/pypa/packaging/pull/907 as well as the type annotation.